int64 index
---
bool success